R/visit.R

```diff
@@ -43,3 +43,166 @@ visit_one <- function(pd_flat, funs) {
   reduce(funs, function(x, fun) fun(x),
          .init = pd_flat)
 }
+
+
+#' Propagate context to terminals
+#'
+#' Implements a very specific pre-visiting scheme, namely to propagate
+#'   indention, spaces and lag_newlines to inner token to terminals. This means
+#'   that information regarding indention, linebreaks and spaces (which is
+#'   relative in `pd_nested`) will be converted into absolute.
+#' @inherit context_towards_terminals
+#' @seealso context_towards_terminals visitors
+context_to_terminals <- function(pd_nested,
```

Comment:
What do we need to change to use this with `pre_visit()` instead of a "manual" recursive call?